bin/propolis-cli/src/main.rs

```diff
@@ -533,12 +533,40 @@ async fn migrate_instance(
     .map(|(role, client, id)| {
         tokio::spawn(async move {
             loop {
-                let state = client
-                    .instance_migrate_status()
-                    .migration_id(migration_id)
-                    .send()
-                    .await?
-                    .state;
+                let state =
+                    client.instance_migrate_status().send().await?.into_inner();
+
+                let migration = if role == "src" {
+                    state.migration_out
+                } else {
+                    state.migration_in
+                };
+
+                // The destination should start reporting migration status as
+                // soon as the ensure request completes. The source may not
+                // have a migration status yet because the request from the
+                // destination needs to arrive first.
+                let Some(migration) = migration else {
+                    if role == "dst" {
+                        anyhow::bail!(
+                            "{role} instance's migration ID wasn't set"
```

Comment:
considered nitpicky: since this is inside an `if role == "dst"`, do we really need to format it here?